src/connectors/stats/connectStats.js

```diff
@@ -88,5 +80,24 @@ export default function connectStats(renderFn, unmountFn = noop) {
     dispose() {
       unmountFn();
     },
+
+    getRenderState(renderState, renderOptions) {
+      return {
+        ...renderState,
+        stats: this.getWidgetRenderState(renderOptions),
+      };
+    },
+
+    getWidgetRenderState({ results, helper }) {
+      return {
+        hitsPerPage: results ? results.hitsPerPage : helper.state.hitsPerPage,
+        nbHits: results ? results.nbHits : 0,
+        nbPages: results ? results.nbPages : 0,
+        page: results ? results.page : helper.state.page || 0,
+        processingTimeMS: results ? results.processingTimeMS : -1,
+        query: results ? results.query : helper.state.query || '',
+        widgetParams,
```

Comment:
I wasn't too sure about the multiple ternaries, should I just go for a simple `if:else` statement?

Reply:
I think that will be cleaner & slightly faster